libd/src/d/semantic/declaration.d

```diff
@@ -23,6 +23,7 @@ enum AggregateType {
 	Union,
 	Struct,
 	Class,
+	Interface,
```

Comment:
I don't think you need a new type. You need the same behavior as for classes. You may want to rename Class to Polymorphic in there.